File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int len = nums.size();
        if(len == 1) return true;
        
        bool dp[len];
        memset(dp, false, sizeof(dp));
        //for(int j=0; j<len; j++) dp[j] = false;
        dp[len-1] = true;
        int last = len-1;
        int i = last;
        
        while(i--) {
            if(nums[i] >= last - i) {
                dp[i] = true;
                last = i;
            }
        }
        return dp[0];
    }
};

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        l = len(nums)
        dp = [False] * l
        dp[l-1] = True
        last = l-1
        i = last-1
        
        while i >= 0:
            if nums[i] >= last-i:
                dp[i] = True
                last = i
            i -= 1
        return dp[0]